File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        int rs = 0, n = nums.length;
        Map<Integer, Integer> fo = new HashMap<>();
        fo.put(0, -1);
        for (int j = 0; j < n; j++) {
            int i = nums[j];
            rs += i;
            rs %= k;
            if (!fo.containsKey(rs)) fo.put(rs, j);
            else if (j - fo.get(rs) >= 2) return true;
        }
        return false;
    }
}

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        mp = {0:-1}
        s = 0
        for i in range(len(nums)):
            s += nums[i]
            if s % k not in mp:
                mp[s%k] = i
            elif i - mp[s%k] > 1:
                return True
        return False
        